### PR TITLE
Uefi format

### PR DESCRIPTION
**Pull request by MikhailKrichanov**

### CODE REVIEW

File: MdeModulePkg/Core/Dxe/Image/Image.c [hunk 1]
```diff
@@ -1189,22 +1221,30 @@ CoreLoadImageCommon (
...
   //
   Status = UefiImageInitializeContextPreHash (
              &ImageContext,
              FHand.Source,
              (UINT32) FHand.SourceSize,
-             ImageIsFromFv
+             UEFI_IMAGE_SOURCE_FV,
+             ImageOrigin
              );
+  if (Status == EFI_NOT_STARTED) {
```
Comment (mikebeaton) [inline MdeModulePkg/Core/Dxe/Image/Image.c:1241]: Should this be EFI_UNSUPPORTED: https://github.com/acidanthera/audk/pull/58#issuecomment-1816474965

File: MdeModulePkg/Core/Dxe/Image/Image.c [hunk 2]
```diff
@@ -1189,22 +1221,30 @@ CoreLoadImageCommon (
...
+    goto Done;
+  }
+
   if (EFI_ERROR (Status)) {
-    ASSERT (FALSE);
-    return Status;
+    CpuDeadLoop ();
+    return Status; ///< Should be unreachable.
```
Comment (mikebeaton) [inline MdeModulePkg/Core/Dxe/Image/Image.c:1247]: Although `return Status;` should be unreachable, if we have it at all should it possibly be `goto Done;`? (With the aim, I guess, of doing the sanest thing we would do, if we were unable to deadloop.)

File: MdeModulePkg/Core/Dxe/Image/Image.c [hunk 3]
```diff
@@ -1259,13 +1299,13 @@ CoreLoadImageCommon (
...
-    Image  = NULL;
     goto Done;
   }
 
   Status = UefiImageInitializeContextPostHash (&ImageContext);
-  if (RETURN_ERROR (Status)) {
-    goto Done;
+  if (EFI_ERROR (Status)) {
+    CpuDeadLoop ();
+    return Status; ///< Should be unreachable.
```
Comment (mikebeaton) [inline MdeModulePkg/Core/Dxe/Image/Image.c:1308]: Also `goto Done;` as above?

File: MdeModulePkg/Core/Dxe/Image/Image.c (hunk 2 above)
Comment (MikhailKrichanov) [reply MdeModulePkg/Core/Dxe/Image/Image.c]: Yes, you're right.

File: MdeModulePkg/Core/Dxe/Image/Image.c (hunk 3 above)
Comment (MikhailKrichanov) [reply MdeModulePkg/Core/Dxe/Image/Image.c]: True.

File: MdeModulePkg/Core/Dxe/Image/Image.c (hunk 1 above)
Comment (MikhailKrichanov) [reply MdeModulePkg/Core/Dxe/Image/Image.c]: It is distinguished deliberately according to policy https://github.com/acidanthera/audk/blob/uefi-format/MdePkg/Library/BasePeCoffLib2/PeCoffInit.c#L484-L503

Comment (mikebeaton) [reply MdeModulePkg/Core/Dxe/Image/Image.c]: Yes, apologies, I misread UefiImageInitializeContextPreHash and thought it was only returning unsupported or success.

File: MdePkg/Library/BaseUefiImageLib/UefiImageLib.c [hunk 4]
```diff
@@ -124,16 +138,31 @@ UefiImageInitializeContextPreHash (
...
+    Status = InternalInitializeContextPreHash (
+               Context,
+               FileBuffer,
+               FileSize,
+               UefiImageFormatUe,
+               ImageOrigin
+               );
+    if (!RETURN_ERROR (Status)) {
+      Context->FormatIndex = UefiImageFormatUe;
+      return RETURN_SUCCESS;
```
Comment (mikebeaton) [inline MdePkg/Library/BaseUefiImageLib/UefiImageLib.c:155]: Making another very minor point, either this and the following early `return` are not really needed (behaviour would be the same without them) or, if they are not removed, the `RETURN_ERROR (Status)` check just below is not really needed (same).

Comment (MikhailKrichanov) [reply MdePkg/Library/BaseUefiImageLib/UefiImageLib.c:155]: I agree, but let's also ask the author. @mhaeuser, what is your opinion?

Comment (mhaeuser) [reply MdePkg/Library/BaseUefiImageLib/UefiImageLib.c:155]: Right, please remove the superfluous check, then. I think it’s an oversight from adding the FormatIndex stuff later.

Comment (mikebeaton) [reply MdePkg/Library/BaseUefiImageLib/UefiImageLib.c:155]: I'd (still) prefer to remove the early returns, as this makes it really obvious when reading the code that the statuses from `InternalInitializeContext(Pre|Post)Hash` are returned - but ofc can go with the decision of the original author, if he doesn't prefer that. :)

Comment (MikhailKrichanov) [reply MdePkg/Library/BaseUefiImageLib/UefiImageLib.c:155]: After careful consideration of the proposed solutions I decided to let @vit9696 have the upper hand.

Comment (vit9696) [reply MdePkg/Library/BaseUefiImageLib/UefiImageLib.c:155]: Better to remove returns in my opinion, this increases readability.

**Comment by vit9696** [comment]
In my opinion, merge this.

File: MdePkg/Library/BaseUefiImageLib/UefiImageLib.c (hunk 4 above)
Comment (mhaeuser) [reply MdePkg/Library/BaseUefiImageLib/UefiImageLib.c:155]: How? You need to branch on the success condition anyway (this was added later and is probably why that check was below) and then you have to keep tracing down to see nothing else is happening.

**Pull request merged**